web/src/user_group_edit.ts

```diff
@@ -111,6 +111,24 @@ export function get_edit_container(group: UserGroup): JQuery {
     );
 }
 
+export function update_group_creation_ui(): void {
+    const $left_panel_icon_button = $("#add_new_user_group .create_user_group_button");
+    const $right_panel_permission_text = $("#groups_overlay .right .creation-permission-text");
+    if (settings_data.user_can_create_user_groups()) {
```

Comment:
We could have realm plan check in `user_can_create_user_groups` but currently the logic to decide when to hide and when to disable the element is somewhat confusing with respect to what to do when user has permission and when plan is limited. Maybe we can decide to have somewhat simple behavior and then can do that refactor.